source/versioned-api/versioned-api.rst

```diff
@@ -0,0 +1,277 @@
+compatibility with MongoDB, both the driver and the server behavior must be
+taken into account.
+
+An application can specify the server API version when creating MongoClient.
+When this is done:
+
+- The client sends the specified API version to the server, causing the server
+  to behave in a manner compatible with that API version.
+- The driver will behave in a manner compatible with the driver's behavior as of
+  the release when the driver first started to support the specified server API
+  version.
+
+Presently there is no specification for how a driver must behave when a
+particular server API version is requested, or what driver operations are
+subject to API compatibility guarantees. Such requirements may be stipulated in
+subsequent specifications.
+
+This specification requires MongoClient to validate that it supports the
+specified server API version, if any, but does not define what such support
+means.
+
+
+``MongoClient`` changes
+-----------------------
+
+``MongoClient`` instances accept a new ``serverApi`` option to allow the user to
+declare an API version:
+
+.. code:: typescript
+
+   class MongoClient {
+       MongoClient(... serverApi: ServerApi);
+   }
+
+   enum ServerApiVersion {
+       v1 = "1",
+   }
+
+   class ServerApi {
+       version: string|ServerApiVersion;
+       strict: Optional<Boolean>; // Default false
+       deprecationErrors: Optional<Boolean>; // Default false
+   }
+
+Drivers SHOULD group the ``serverApi`` option with other similar client options
+like ``autoEncryptionOpts``. Drivers MUST NOT allow specification of any
+versioned API options via the connection string. See the
+`design rationale <_rationale_no_uri_options>`_ for more details.
+
+
+ServerApiVersion enumeration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This specification and subsequent specifications will define the known API
+versions. Drivers SHOULD define an enumeration containing the known API
+versions, using the version identifiers given. Drivers MAY deviate from the
+version identifiers used in this and subsequent specifications if doing so is
+necessary given the driver's programming language's constraints.
+
+Drivers for languages that don't have enums (e.g. PHP) MUST expose the version
+as a string, but SHOULD offer constants to allow for IDE features such as code
+completion. In these cases, the driver MUST validate (e.g. when the application
+provides a version string to the ``ServerApi`` class) that the version string is
+valid and trigger a client-side error if an unknown API version was used.
+
+
+ServerApi class
+~~~~~~~~~~~~~~~
+
+The ``ServerApi`` class stores an API version, along with flags that decide
+whether or not unknown or deprecated commands in the specified API version
+trigger a server-side error. A ``version`` MUST be specified when declaring an
+API version, while the ``strict`` and ``deprecationErrors`` options are both
+optional and default to ``false``. The ``ServerApi`` class is considered
```

Comment:
> are both optional and default to ``false``

If it's optional then it defaults to `unset` or `not present`.  And it contradicts what is said below:

> Drivers SHOULD NOT add these options if the user did not specify a value.

Reply:
Do you mean to say:

> are either optional, or non-optional and default to ``false``

?

Reply:
Removed any mention of a default value corresponding with the above change requiring drivers to not send apiStrict and apiDeprecationErrors unless they were explicitly specified.